### PR TITLE
fix(select): communicate component state to screen reader

### PR DESCRIPTION
**Pull request by chasestarr**
#### Description

Improves the select component so that screen readers can successfully operate it. See https://github.com/uber/baseweb/pull/2929 for an approach that would make `Select` a semantic combobox, but may lead to unintended changes in behavior. This Diff seems to be a good middle ground between making it operational, but not changing functionality.

#### Scope

- [x] Patch: Bug Fix

### CODE REVIEW

File: src/select/types.js [hunk 1]
```diff
@@ -179,6 +179,7 @@ export type PropsT = {
 };
 
 export type SelectStateT = {
+  activeDescendant: ?string,
```
Comment (gergelyke) [inline src/select/types.js]: should we add these to the TS type definitions too?

**Comment by tajo** [comment]
This works pretty great overall. However; it seems that backspace doesn't work anymore.

**Comment by chasestarr** [comment]
> This works pretty great overall. However; it seems that backspace doesn't work anymore.

Can you please provide some reproduction steps and also validate against master? I didn't make any changes related to backspace in this PR. It may be a separate regression

**Comment by nadiia** [comment]
> This works pretty great overall. However; it seems that backspace doesn't work anymore.

Do you mean to remove the selected value? It only works for `multi` I think.

**Comment by tajo** [comment]
Ah, it's not a regression. When you select a value you are not able to delete part of it to search for other options. That's a bit annoying.

**Pull request merged**